cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -926,133 +926,152 @@ static PyMethodDef gpi_sim_hdl_methods[] = {
         (PyCFunction)get_signal_val_long, METH_NOARGS, PyDoc_STR(
             "get_signal_val_long($self)\n"
             "--\n\n"
-            "Get the value of a signal as a long"
+            "get_signal_val_long(self) -> int\n"
+            "Get the value of a signal as an integer."
         )
     },
     {"get_signal_val_str",
         (PyCFunction)get_signal_val_str, METH_NOARGS, PyDoc_STR(
             "get_signal_val_str($self)\n"
             "--\n\n"
-            "Get the value of a signal as an ASCII string"
+            "get_signal_val_str(self) -> str\n"
+            "Get the value of a signal as a string."
         )
     },
     {"get_signal_val_binstr",
         (PyCFunction)get_signal_val_binstr, METH_NOARGS, PyDoc_STR(
             "get_signal_val_binstr($self)\n"
             "--\n\n"
-            "Get the value of a signal as a binary string"
+            "get_signal_val_binstr(self) -> str\n"
+            "Get the value of a logic vector signal as a string of (``0``, ``1``, ``X``, etc.), one element per character."
         )
     },
     {"get_signal_val_real",
         (PyCFunction)get_signal_val_real, METH_NOARGS, PyDoc_STR(
             "get_signal_val_real($self)\n"
             "--\n\n"
-            "Get the value of a signal as a double precision float"
+            "get_signal_val_real(self) -> float\n"
+            "Get the value of a signal as a float."
         )
     },
     {"set_signal_val_long",
         (PyCFunction)set_signal_val_long, METH_VARARGS, PyDoc_STR(
             "set_signal_val_long($self, action, value, /)\n"
```

Comment:
Do _argument_ type annotations work on this line for `inspect.signature`?

Reply:
```
    {"register_timed_callback", register_timed_callback, METH_VARARGS, PyDoc_STR(
        "register_timed_callback(time: int, func, /, *args)\n"
        "--\n\n"
        "Register a timed callback."
    )},
```
```
>>> inspect.signature(simulator.register_timed_callback)
<Signature (time, func, /, *args)>
```
In Sphinx:
```
cocotb.simulator.register_timed_callback()
```

Reply:
No, I manually tried running that function on a function I added annotations to.
```
    {"register_timed_callback", register_timed_callback, METH_VARARGS, PyDoc_STR(
        "register_timed_callback(time: int, func, /, *args)\n"
        "--\n\n"
        "Register a timed callback."
    )},
```
```
>>> from cocotb import simulator
>>> import inspect
>>> inspect.signature(simulator.register_timed_callback)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kaleb/anaconda3/envs/py3.8/lib/python3.8/inspect.py", line 3093, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped)
  File "/home/kaleb/anaconda3/envs/py3.8/lib/python3.8/inspect.py", line 2842, in from_callable
    return _signature_from_callable(obj, sigcls=cls,
  File "/home/kaleb/anaconda3/envs/py3.8/lib/python3.8/inspect.py", line 2296, in _signature_from_callable
    return _signature_from_builtin(sigcls, obj,
  File "/home/kaleb/anaconda3/envs/py3.8/lib/python3.8/inspect.py", line 2109, in _signature_from_builtin
    return _signature_fromstr(cls, func, s, skip_bound_arg)
  File "/home/kaleb/anaconda3/envs/py3.8/lib/python3.8/inspect.py", line 2057, in _signature_fromstr
    p(name, default)
  File "/home/kaleb/anaconda3/envs/py3.8/lib/python3.8/inspect.py", line 2034, in p
    name = parse_name(name_node)
  File "/home/kaleb/anaconda3/envs/py3.8/lib/python3.8/inspect.py", line 1999, in parse_name
    raise ValueError("Annotations are not currently supported")
ValueError: Annotations are not currently supported
```